src/core/ext/transport/chttp2/transport/writing.c

```diff
@@ -322,6 +322,20 @@ grpc_chttp2_begin_write_result grpc_chttp2_begin_write(
                     (send_bytes == s->compressed_data_buffer->length &&
                      s->flow_controlled_buffer.length == 0 &&
                      s->fetching_send_message == NULL);
+                if (is_last_data_frame && s->stream_compression_ctx) {
+                  GPR_ASSERT(grpc_stream_compress(
```

Comment:
Asserts are tricky. When can `grpc_stream_compress` fail? If its failure is recoverable, we shouldn't crash the whole process. Imagine this could end up as a DoS attack against a server, etc.

Reply:
Are you going to do that? I don't see any change.

Reply:
Did here: https://github.com/grpc/grpc/pull/12244/files#diff-b1da1e4e504c2379a66dc09ddfdd1610R331

Reply:
ugh github UI...